src/containers/User/Login.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+import './Registration.css';
+import './Login.css';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { withAlert } from 'react-alert';
+import { loginUser } from '../../api/api';
+import { setAuthToken, setUserId } from '../../global/state/globalActions';
+import { connect } from 'react-redux';
+
+export default class Login extends Component {
+   constructor(props) {
+      super(props);
+
+      this.state = {
+         email: '',
+         password: ''
+      };
+   }
+
+   handleInputChange = (e) => {
+      const { name, value } = e.target;
+      this.setState(() => ({ [name]: value }));
+   };
+
+   handleLoginClick = async () => {
+      try {
+         const { email, password } = this.state;
+         const res = await loginUser({ email, password });
+         this.setState({
+            email: '',
+            password: ''
+         });
+         this.props.setAuthToken(res.data.token);
+         this.props.setUserId(res.data.userId);
+         this.props.history.push('/');
+      } catch (error) {
+         this.props.alert.error('Failed to login. Please check username and password.');
+      }
+   };
+
+   render() {
+      const { email, password } = this.state;
+      return (
+         <div className="Registration">
+            <form className="register-form">
+               <a href='/'><div className="company-logo" /></a>
+               <div className="form-group">
+                  <input
+                     onChange={this.handleInputChange}
+                     type="email"
+                     name="email"
+                     className="email"
+                     value={email}
+                     placeholder="email"
+                  />
+               </div>
+               <div className="form-group">
+                  <input
+                     onChange={this.handleInputChange}
+                     type="password"
+                     name="password"
+                     className="password"
+                     value={password}
+                     placeholder="password"
+                  />
+               </div>
+               <div className="register-btn" onClick={this.handleLoginClick}>
+                  login
+               </div>
+               <div className='login-registration-link'>
+                  <a href='/registration'>Don't have an account?</a>
+               </div>
+            </form>
+         </div>
+      );
+   }
+}
+
+Login.propTypes = {
+   history: PropTypes.object.isRequired,
+   alert: PropTypes.object.isRequired,
+};
+
+const ConnectedLogin = withRouter(withAlert(connect(null, {
+   setAuthToken, setUserId
+})(Login)));
+
+export {
```

Comment:
You can just export withRouter from line 85

Reply:
IN